src/lib/AccordionItem.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+  import { slide } from "svelte/transition";
+  // TODO Make this more generic by taking some of these as props too or forwarding events
+  import {
+    formOpen,
+    mapHover,
+    openFromSidebar,
+    sidebarHover,
+  } from "../stores.js";
+
+  export let id: number;
+  export let label: string;
+
+  $: isOpen = $formOpen == id;
+  const toggle = () => {
+    formOpen.update((current) => {
+      if (current == id) {
+        return null;
+      } else {
+        return id;
+      }
+    });
+    if ($formOpen == id) {
+      // Always set this to null first, to force subscribers to see the update.
+      // It's possible to open something from the sidebar, close it (by
+      // clicking on the map or using the sidebar), then reopen the same thing.
+      openFromSidebar.set(null);
+      openFromSidebar.set(id);
+    }
+  };
+
+  // TODO Design choice: While a form is open and we hover on something else in
+  // the map, should we underline the item in the sidebar? For now, yes.
+  $: underlineRemotely = $mapHover == id;
+
+  let contents: HTMLDivElement;
+  function scroll() {
+    contents?.scrollIntoView({ behavior: "smooth" });
+  }
+</script>
+
+<button
+  on:click={toggle}
+  on:mouseenter={() => sidebarHover.set(id)}
+  on:mouseleave={() => sidebarHover.set(null)}
+  aria-expanded={isOpen}
+  class:underlined={underlineRemotely}
+  ><svg
+    style="tran"
+    width="20"
+    height="20"
+    fill="none"
+    stroke-linecap="round"
+    stroke-linejoin="round"
+    stroke-width="2"
+    viewBox="0 0 24 24"
+    stroke="currentColor"><path d="M9 5l7 7-7 7" /></svg
+  >
+  {label}
+</button>
+{#if isOpen}
+  <div
+    bind:this={contents}
+    transition:slide={{ duration: 100 }}
```

Comment:
The effect may be a little choppy; we spend 100ms sliding the accordion down, then scroll to make sure the form is in view. Any better ideas?